Adding modify tab, renaming old tabs

tlauver opened the pull request:
Editing existing symbol feature functionality using the selection tool in pro, and fixed issue #71

csmoore commented on source/ProSymbolEditor/Models/SelectedFeature.cs:
```diff
@@ -0,0 +1,26 @@
+﻿using ArcGIS.Desktop.Framework.Contracts;
```
Needs default Apache Header

csmoore commented on source/ProSymbolEditor/Models/SymbolAttributes/SymbolAttributeSet.cs:
```diff
@@ -39,6 +39,139 @@ public SymbolAttributeSet()
+
+            if (fieldValues.ContainsKey("indicator"))
+            {
+                //if (!string.IsNullOrEmpty(fieldValues["indicator"]))
```
commented out code

csmoore commented:
I got this warning on startup:
```
System.Windows.Data Error: 40 : BindingExpression path error: 'Name' property not found on 'object' ''SelectedFeature' (HashCode=61628039)'. BindingExpression:Path=Name; DataItem='SelectedFeature' (HashCode=61628039); target element is 'Grid' (Name=''); target property is 'ToolTip' (type 'Object')
System.Windows.Data Error: 40 : BindingExpression path error: 'PreviewImage' property not found on 'object' ''SelectedFeature' (HashCode=61628039)'. BindingExpression:Path=PreviewImage; DataItem='SelectedFeature' (HashCode=61628039); target element is 'Image' (Name=''); target property is 'Source' (type 'ImageSource')
```

csmoore commented:
I don't know if it is related to this PR, but it seems to be taking a long time to add features to the map now if I set the "Content" attribute - can you see if you can repro this odd behavior?

![image](https://cloud.githubusercontent.com/assets/3090809/15781201/ffe5f136-2972-11e6-8f1d-8b0f4e976df8.png)

csmoore commented on source/ProSymbolEditor/Models/SymbolAttributes/SymbolAttributeSet.cs:
```diff
@@ -309,7 +442,139 @@ public void PopulateRowBufferWithAttributes(ref RowBuffer rowBuffer)
+        {
+            if (!string.IsNullOrEmpty(DisplayAttributes.Identity))
+            {
+                feature["identity"] = DisplayAttributes.Identity;
```
Given the number of places these are used - I'm wondering if these attribute names should be variables/constants defined once - in case we need to change the name -or- they will differ 2525D to C to B

csmoore replied:
Note: this was just a (low priority) observation - no need to change with this PR

csmoore commented on source/ProSymbolEditor/ViewModels/MilitarySymbolDockpaneViewModel.cs:
```diff
@@ -590,6 +665,74 @@ private void ActivateDrawFeatureSketchTool(object parameter)
+                }
+                catch (Exception exception)
+                {
+
```
Some meaningful trace/warning message here would be helpful

csmoore commented on source/ProSymbolEditor/ViewModels/MilitarySymbolDockpaneViewModel.cs:
```diff
@@ -804,77 +947,122 @@ private void CopyImageToClipboard(object parameter)
         }
 
-        private void LoadSymbolFile(object parameter)
+        private void LoadSymbolFile(bool isEditSymbol)
```
Sorry, I don't quite understand this method/name - can you change to a more meaningful name or add some comments about what method generally does (ex: sets symbol model state from...? and perhaps how edit case is different).

csmoore commented:
Finished reviewing - looks good - my last comments/suggestions pertain to the new Modify tab - to see if it would be readily/easily possible to change any of these items with PR before end of sprint (if not, I can just write issues for).

`1.` Can we add the select tool to this tab? If not, can we add text to tell the user to use the select tool to select a symbol (I think for now users might be confused about how to "pick" a symbol) 

![image](https://cloud.githubusercontent.com/assets/3090809/15781846/e827d78c-2975-11e6-99a2-7942d6cb7c32.png)

`2.` Can we auto-select the first selection/list-item so the Modify UI shows this symbol without having to then select the list item 

`3.` Can we add more description to the picked symbols (ex. Layer : Symbol Set(Name) : Entity(Name) ) to make this name more user-friendly 

![image](https://cloud.githubusercontent.com/assets/3090809/15782069/cc303b2c-2976-11e6-9645-87845ad5d8dc.png)

tlauver commented:
I doubt we can do all of these before the end of the sprint, since it ends today and you've mentioned some other items to look at.

1. Should be possible, but will require a bit of work.
2. Again, should be possible.
3. Possible, but we'll have to see how it impacts performance as I'll have to query the data in each selected feature layer to get that data.

csmoore commented:
No biggie - for now (as part of PR), can you at least add static text to Modify Tab to let the user know to "use the select tool to select a feature to modify (or something along these lines)" - I'll issues for other enhancements

tlauver commented:
Updated the code with the issues discussed above (Besides those 3 items - which I'll start looking at now).  I tested the Context attribute (and the other ones) but never saw a slow down of editing the database.  Maybe something else going on?

tlauver commented:
@csmoore If I can change it to what you suggested above:

> 3. Can we add more description to the picked symbols (ex. Layer : Symbol Set(Name) : Entity(Name) ) to make this name more user-friendly

Then the user would not know which feature is which, since their is no differentiation between different features if they're the same symbolset.  Is that okay?  Or should I still include Object Id in there?

csmoore commented:
Including the OID at the end sounds fine - mainly just wanted something human readable included to give a user a better hint at what they had picked

csmoore commented:
@tlauver - sorry I haven't merged btw - I had been waiting to get a chance to writeup the issues above before I merged/closed

tlauver commented:
Alright, those three issues are now fixed.  Added a new selection button (that mimics the one in Pro), auto selected first item when selection is done, and updated the description of each item.  Also added that if a user clicks on a result item, the feature will flash on the map - hopefully helping to locate it.

csmoore commented:
Cool (good idea on the flash) - Many thanks - I test out first thing tomorrow

csmoore commented:
Modify additions look/test good. Thanks for adding that & updated doc also.
